datetime_test.go

```diff
@@ -119,79 +119,125 @@ func TestDateTime_Getters(t *testing.T) {
 }
 
 func TestDateTime_IsEqual(t *testing.T) {
-	reporter := newMockReporter(t)
+	cases := []struct {
+		name        string
+		time        time.Time
+		value       time.Time
+		wantIsEqual chainResult
+	}{
+		{
+			name:        "equal to value",
+			time:        time.Unix(0, 1234),
+			value:       time.Unix(0, 1234),
+			wantIsEqual: success,
+		},
+		{
+			name:        "not equal to value",
+			time:        time.Unix(0, 1234),
+			value:       time.Unix(0, 4321),
+			wantIsEqual: failure,
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			reporter := newMockReporter(t)
 
-	value := NewDateTime(reporter, time.Unix(0, 1234))
+			NewDateTime(reporter, tc.time).
+				IsEqual(tc.value).chain.assert(t, tc.wantIsEqual)
```

Comment:
nitpick: we usually use a bit different formatting:

```go
NewDateTime(reporter, tc.time).IsEqual(tc.value).
    chain.assert(t, tc.wantIsEqual)
```

It's a bit easier to read because the important part (`IsEqual(tc.value)`) is more noticeable. Could you please adjust formatting here and in other modified tests?